### PR TITLE
[ONNX] Enable Pad operator and batch normalization non-spatial mode

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/batch_norm.cpp [hunk 1]
```diff
@@ -34,7 +47,8 @@ namespace ngraph
                 NodeVector batch_norm(const Node& node)
                 {
                     NodeVector inputs{node.get_ng_inputs()};
-                    auto x = inputs.at(0);
+                    auto data = inputs.at(0);
+                    const Shape& data_shape = data->get_shape();
```
[the remark this answers is not in the document]
Comment (postrational) [reply src/ngraph/frontend/onnx_import/op/batch_norm.cpp]: +1

File: src/ngraph/frontend/onnx_import/op/batch_norm.cpp [hunk 2]
```diff
@@ -47,18 +61,46 @@ namespace ngraph
...
-                        return {std::make_shared<ngraph::op::BatchNormInference>(
-                            epsilon, scale, bias, x, mean, var)};
+                        auto mean =
+                            legacy_style_broadcast_for_binary_operation(data, inputs.at(3), 1)
+                                .at(1);
+                        auto variance =
+                            legacy_style_broadcast_for_binary_operation(data, inputs.at(4), 1)
+                                .at(1);
+
+                        bias = legacy_style_broadcast_for_binary_operation(data, bias, 1).at(1);
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/batch_norm.cpp]: Do we need to mix `legacy_style_broadcast_for_binary_operation` and `numpy_style_broadcast_for_binary_operation` in this implementation? Would it be possible to stick to just one of them?

File: src/ngraph/frontend/onnx_import/op/batch_norm.cpp [hunk 3]
```diff
@@ -47,18 +61,46 @@ namespace ngraph
...
 
-                    if (inputs.size() >= 5)
+                    if (spatial == 0)
                     {
-                        mean = inputs.at(3);
-                        var = inputs.at(4);
-                        return {std::make_shared<ngraph::op::BatchNormInference>(
-                            epsilon, scale, bias, x, mean, var)};
+                        auto mean =
+                            legacy_style_broadcast_for_binary_operation(data, inputs.at(3), 1)
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/batch_norm.cpp]: It would make sense to turn this into a helper function `legacy_style_broadcast_to(input, new_shape)`?

File: src/ngraph/frontend/onnx_import/op/batch_norm.cpp [hunk 4]
```diff
@@ -47,18 +61,46 @@ namespace ngraph
...
+                                .at(1);
+                        auto variance =
+                            legacy_style_broadcast_for_binary_operation(data, inputs.at(4), 1)
+                                .at(1);
+
+                        bias = legacy_style_broadcast_for_binary_operation(data, bias, 1).at(1);
+                        scale = legacy_style_broadcast_for_binary_operation(data, scale, 1).at(1);
+
+                        std::shared_ptr<ngraph::Node> epsilon_node = ngraph::op::Constant::create(
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/batch_norm.cpp]: Perhaps a helper function `create_constant_node(double input, new_shape)`?

Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/batch_norm.cpp]: Data type and constant node data type can be different so there will be still 3 arguments.

**Comment by postrational** [comment]
Please use Adam's helper to make constants.
